2019-08-03-q-and-as-about-nix-for-haskellers.md

```diff
@@ -0,0 +1,692 @@
+title = Questions and Answers about Nix for Haskellers
+tags = haskell, nixos
+summary = Overview of Nix for Haskell development
+draft = true
+
+------------------------------------------------------
+
+Last week Dmitrii Kovanikov ([@ChShersh](https://twitter.com/ChShersh)) posted
+a [request](https://twitter.com/ChShersh/status/1154782785967542272?s=20) on
+Twitter asking if anyone had the time to answer some questions about how to
+understand Nix as it applies to Haskell programming.
+
+Dmitrii is working on a scaffolding tool for Haskell projects called
+[summoner](https://github.com/kowainik/summoner).  There is an
+[open PR](https://github.com/kowainik/summoner/pull/283) adding Nix support.
+However, Dmitrii doesn't have any Nix experience, so he'd like to learn more
+about using Nix for Haskell development before tackling the PR.
+
+I've been using Nix for about a year and half, both at work and on my own
+projects.  I think there is a big lack of documentation surrounding Nix and
+Haskell, so I told him I'd be able to answer any questions he had.
+
+He wrote me back with a _great_ list of questions.  I thought it would be an
+injustice to the Haskell community if I didn't share my answers, so I decided
+to turn his questions and my answers into an article.
+
+This article is mainly for Haskellers that have heard about Nix (on Twitter,
+Reddit, YouTube, etc), know about its advantages, but haven't actually played
+around with it.  It is mainly a **high-level overview** of how Nix works with
+Haskell.  It is not a hands-on tutorial or walk-through.
+
+The first section of this article is an intro to Nix, followed by answers
+to each of Dmitrii's questions.[^7]
+
+## Intro
+
+You may have heard that Nix is complicated and hard to learn.  In my
+experience, this is true.
+
+It is similar to Haskell.  The learning curve is steep, but once you get the
+hang of it, it is very powerful and flexible.
+
+*What makes it so hard?*
+
+To start off with, Nix refers to two things:
```

Comment:
Probably you can also give a link to this small blog post:

* https://typeclasses.com/nix/intro